cpp/src/arrow/compute/exec.cc

```diff
@@ -334,23 +334,20 @@ struct NullGeneralization {
     if (datum.type()->id() == Type::NA) {
       return ALL_NULL;
     }
-
```

Comment:
Note one might want to add:
```c++
  if (!HasValidityBitmap(datum.type()->id()))  {
    return ALL_VALID;
  }
```

This will only have an effect for union types, so not very important though.